File: IrrgartenP2/irrgartenRuby/monster.rb
# monster.rb
class Monster
  INITIAL_HEALTH = 3

  attr_reader :health, :strength, :intelligence, :row, :col

  def initialize(strength, intelligence)
    @health = INITIAL_HEALTH
    @strength = strength
    @intelligence = intelligence
    @row = -1
    @col = -1
  end

  def dead?
    @health <= 0
  end

  def attack
    Dice.intensity(@strength)
  end

  def set_pos(row, col)
    @row = row
    @col = col
  end

  def got_wounded
    @health -= 1
  end

  def to_s
    "Monster[Health: #{@health}, Strength: #{@strength}, Intelligence: #{@intelligence}, Position: (#{@row}, #{@col})]"
  end

  def defend(received_attack)
    raise NotImplementedError, "Method not implemented yet."
  end
end

File: IrrgartenP2/IrrgartenJava/src/irrgartenjava/Monster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgartenjava;

/**
 *
 * @author moham
 */
// Monster.java
public class Monster {
    private static final int INITIAL_HEALTH = 3;
    private int health;
    private float strength;
    private float intelligence;
    private int row; 
    private int col; 

    public Monster(float strength, float intelligence) {
        this.health = INITIAL_HEALTH;
        this.strength = strength;
        this.intelligence = intelligence;
        this.row = -1; // Initialize to an invalid position
        this.col = -1; // Initialize to an invalid position
    }

    public boolean dead() {
        return health <= 0;
    }

    public float attack() {
        return Dice.intensity(strength);
    }

    public void setPos(int row, int col) {
        this.row = row;
        this.col = col;
    }

    @Override
    public String toString() {
        return "Monster[Health: " + health + ", Strength: " + strength + ", Intelligence: " + intelligence + ", Position: (" + row + ", " + col + ")]";
    }

    public void gotWounded() {
        health--;
    }

    public boolean defend(float receivedAttack) {
        throw new UnsupportedOperationException("Method not implemented yet.");
    }
}
